src/api/app/services/trigger_controller_service/scm_extractor.rb

```diff
@@ -16,19 +18,30 @@ class SCMExtractor
       }
     }.freeze
 
+    # FIXME: This is a factory
     def initialize(scm, event, webhook_payload)
-      # TODO: What should we do when the user sends a wwwurlencoded payload? Raise an exception?
-      @webhook_payload = webhook_payload.deep_symbolize_keys
       @scm = scm
       @event = event
+      # TODO: What should we do when the user sends a wwwurlencoded payload? Raise an exception?
+      @webhook_payload = webhook_payload.deep_symbolize_keys
+      @extractor = SCM_EXTRACTORS.dig(@scm, @event)
     end
 
     # TODO: What happens when some of the keys are missing?
     def call
-      extractor = SCM_EXTRACTORS.dig(@scm, @event)
       return unless extractor
 
       SCMWebhook.new(payload: extractor.new(@webhook_payload).payload)
     end
+
+    def valid?
+      @scm.present? && @extractor.present?
+    end
+
+    def error_message
+      return 'Only GitHub, GitLab and Gitea are supported. Could not find the required HTTP request headers X-GitHub-Event, X-Gitlab-Event or X-Gitea-Event.' if @scm.nil?
+
+      'This SCM event is not yet supported' if @extractor.nil?
```

Comment:
```suggestion
      'This SCM event is not supported' if @extractor.nil?
```

Reply:
@danidoni most likely we will never support it, so I would remove the `yet` part

Reply:
Removed